pkg/protocol/bthrift/apache/apache.go

```diff
@@ -17,11 +17,43 @@
 package apache
 
 import (
+	"errors"
+
 	"github.com/apache/thrift/lib/go/thrift"
 	"github.com/cloudwego/gopkg/protocol/thrift/apache"
 )
 
 func init() {
 	// it makes github.com/cloudwego/gopkg/protocol/thrift/apache works
-	_ = apache.RegisterNewTBinaryProtocol(thrift.NewTBinaryProtocol)
+	apache.RegisterCheckTStruct(checkTStruct)
```

Comment:
Previously, it was for registering the implementation of TBinaryProtocol, but now I feel that the necessity of registration is not very high? Call directly? Reflection aims to remove Apache dependencies without using assertion calls, but now it's still assertion, as if registration is not necessary